File: 2302-count-subarrays-with-score-less-than-k/2302-count-subarrays-with-score-less-than-k.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        j=0
        sum=0
        count=0
        for i in range(0,len(nums)):
            sum+=nums[i]
            while sum*(i-j+1)>=k:
                sum-=nums[j]
                j+=1
            count+=(i-j+1)
        return count    
        
        

File: 2302-count-subarrays-with-score-less-than-k/2302-count-subarrays-with-score-less-than-k.cpp
class Solution {
public:
    long long countSubarrays(vector<int>& nums, long long k) 
    {
        int n=nums.size();
        long long count=0;
        int j=0;
        long long sum=0;
        for(int i=0;i<n;i++)
        {
            sum+=nums[i];
            while((1ll*sum*(i-j+1))>=k)
            {
                sum-=nums[j];
                j++;
            }
            count+=(i-j+1);
        }
        return count;
    }
};